core/shared/src/main/scala/zio/FiberRefs.scala

```diff
@@ -181,4 +181,95 @@ object FiberRefs {
 
   private[zio] def apply(fiberRefLocals: Map[FiberRef[_], ::[(FiberId.Runtime, Any)]]): FiberRefs =
     new FiberRefs(fiberRefLocals)
+
+  /**
+   * A `Patch` captures the changes in `FiberRef` values made by a single fiber
+   * as a value. This allows fibers to apply the changes made by a workflow
+   * without inheriting all the `FiberRef` values of the fiber that executed the
+   * workflow.
+   */
+  sealed trait Patch extends Function2[FiberId.Runtime, FiberRefs, FiberRefs] { self =>
+    import Patch._
+
+    /**
+     * Applies the changes described by this patch to the specified collection
+     * of `FiberRef` values.
+     */
+    def apply(fiberId: FiberId.Runtime, fiberRefs: FiberRefs): FiberRefs = {
+
+      @tailrec
+      def loop(fiberRefs: FiberRefs, patches: List[Patch]): FiberRefs =
+        patches match {
+          case Add(fiberRef, value) :: patches =>
+            loop(fiberRefs.updatedAs(fiberId)(fiberRef, value), patches)
+          case AndThen(first, second) :: patches =>
+            loop(fiberRefs, first :: second :: patches)
+          case Empty :: patches =>
+            loop(fiberRefs, patches)
+          case Remove(fiberRef) :: patches =>
+            loop(fiberRefs.delete(fiberRef), patches)
+          case Update(fiberRef, patch) :: patches =>
+            loop(
+              fiberRefs.updatedAs(fiberId)(fiberRef, fiberRef.patch(patch)(fiberRefs.getOrDefault(fiberRef))),
+              patches
+            )
+          case Nil =>
+            fiberRefs
+        }
+
+      loop(fiberRefs, List(self))
+    }
+
+    /**
+     * Combines this patch and the specified patch to create a new patch that
+     * describes applying the changes from this patch and the specified patch
+     * sequentially.
+     */
+    def combine(that: Patch): Patch =
+      AndThen(self, that)
+  }
+
+  object Patch {
+
+    /**
+     * The empty patch that describes no changes to `FiberRef` values.
+     */
+    val empty: Patch =
+      Patch.Empty
+
+    /**
+     * Constructs a patch that describes the changes between the specified
+     * collections of `FiberRef`
+     */
+    def diff(oldValue: FiberRefs, newValue: FiberRefs): Patch = {
```

Comment:
Does it make sense to put this functionality in a Differ?

Reply:
I think the presence of `FiberId` in the signature of `apply` makes it not quite fit the `Differ` interface.